File: workspace/solutions/ecmascript/src/s0/s0/s0/s1/s_0001_00.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {[number, number] | void}
 */
const twoSum = function (nums, target) {
    const map = {};
    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        const diff = target - num;
        const j = map[diff];
        if (j != null) {
            return [j, i];
        }
        map[num] = i;
    }
};
// @lc code=end

export default twoSum;

File: workspace/solutions/kotlin/lib/src/main/kotlin/s0/s0/s0/s1/s_0001_00/s_0001_00.kt
/*
 * @lc app=leetcode.cn id=1 lang=kotlin
 *
 * [1] 两数之和
 */

package s0.s0.s0.s1.s_0001_00

// @lc code=start
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        nums.forEachIndexed { i, num ->
            val diff = target - num
            val j = map.get(diff)
            if (j != null) {
                return intArrayOf(j, i)
            }
            map[num] = i
        }
        return intArrayOf()
    }
}
// @lc code=end
